src/prdoc.rs

```diff
@@ -196,9 +196,31 @@ fn compare_deps(
         .and_then(|d| d.as_table())
         .unwrap_or(&t);
 
-    for (name, _) in deps {
-        let (_old_pkg, old_dep, old_root_dep) = get_dep(workspace, name, old, old_root)?;
-        let (new_pkg, new_dep, new_root_dep) = get_dep(workspace, name, new, new_root)?;
+    for name in deps.iter().map(|(name, _)| name).collect::<HashSet<_>>() {
```

Comment:
As this is iterating over the dependency table of the manifest I don't see where there are duplicates that a hashmap would help.

If this does some how does lead to a speed up I'd still prefer not to use a hashmap as it leads to non determinism when things go wrong which I'd like to avoid.

Reply:
oh, you're right, I double-checked again and this is not an issue, I reverted it